src/configs/nuage/scripts/pps-simulator/simulate_pps_stats.py

```diff
@@ -472,90 +486,111 @@ def __init__(self, flowData):
+            probe_record = {}
+            for s_nsg_ind, src_nsg in enumerate(nsgs):
+                initPacket = 1
+                d_nsg_ind = s_nsg_ind + 1
+                while d_nsg_ind < len(nsgs):
+                    # d_nsg_ind = s_nsg_ind
+                    # while d_nsg_ind == s_nsg_ind:
+                    #     d_nsg_ind = random.randrange(0, len(nsgs) - 1)
+                    dest_nsg = nsgs[d_nsg_ind]
+                    srcUp = randint(0, 1)
+                    destUp = randint(0, 1)
+                    domain = randint(0, len(domains) - 1)
+                    npm_grp = random.choice(npm_grps)
+                    duc_grp = random.choice(self.duc_grps)
+                    perf_mon = npm_grp["perf_monitor"]
+                    avg_latency = random.uniform(0.01, 1000)
+                    avg_jitter = random.uniform(0.01, 100)
+                    avg_pktloss = random.uniform(0.00, 100)
+                    probe_record["timestamp"] = long(timestamp)
+                    probe_record["SrcNSG"] = src_nsg["nsg_id"]
+                    probe_record["SourceNSG"] = src_nsg["nsg_name"]
+                    probe_record["DstNSG"] = dest_nsg["nsg_id"]
+                    probe_record["DestinationNSG"] = dest_nsg["nsg_name"]
+                    probe_record["SrcUplink"] = srcuplinks[srcUp]
+                    probe_record["DstUplink"] = destuplinks[destUp]
+                    probe_record["NPMGroup"] = npm_grp["npmgrp_name"]
+                    probe_record["PerfMonitor"] = perf_mon["perfmon_name"]
+                    probe_record["MonitorServiceClass"] = perf_mon["service_class"]
+                    probe_record["MonitorPayload"] = perf_mon["monitor_payload"]
+                    probe_record["MonitorProbeInterval"] = perf_mon["probe_interval"]
+                    probe_record["MonitorProbeNoOfPackets"] = perf_mon["probe_num_pkts"]
+                    probe_record["AvgDelay"] = avg_latency
+                    probe_record["AvgJitter"] = avg_jitter
+                    probe_record["AvgPktLoss"] = avg_pktloss
+                    probe_record["Domain"] = domains[domain]
+                    probe_record["EnterpriseName"] = self.def_ent_name
+                    probe_record["ControlSessionState"] = self.contro_states[0]
+                    underlayId = random.randrange(0, 10)
+                    probe_record["UnderlayID"] = underlayId
+                    probe_record["UnderlayName"] = "Underlay" + str(
+                        underlayId)
+                    probe_record["DUCGroupID"] = duc_grp["duc_id"]
+                    probe_record["DUCGroup"] = duc_grp["duc_name"]
+
+                    if self.firstTS and initPacket == 1:
+                        probe_record["ControlSessionState"] = self.contro_states[2]
+                        probe_record["AvgDelay"] = "Null"
+                        probe_record["AvgJitter"] = "Null"
+                        probe_record["AvgPktLoss"] = "Null"
```

Comment:
"properties" : {
          "AvgDelay" : {
            "type" : "double"
          },
          "AvgJitter" : {
            "type" : "double"
          },
          "AvgPktLoss" : {
            "type" : "double"
          },

This can not be a string and double together. I am getting a 

elasticsearch.helpers.BulkIndexError: (u'2 document(s) failed to index.', [{u'create': {u'status': 400, u'_type': u'nuage_doc_type', u'_id': u'AV5_F8ectxRjxMV1WI9H', u'error': {u'caused_by': {u'reason': u'For input string: "Null"', u'type': u'number_format_exception'}, u'reason': u'failed to parse [AvgPktLoss]', u'type': u'mapper_parsing_exception'}, u'_index': u'nuage_dpi_probestats_2017_09_14'}}, {u'create': {u'status': 400, u'_type': u'nuage_doc_type', u'_id': u'AV5_F8ectxRjxMV1WI-A', u'error': {u'caused_by': {u'reason': u'For input string: "Null"', u'type': u'number_format_exception'}, u'reason': u'failed to parse [AvgPktLoss]', u'type': u'mapper_parsing_exception'}, u'_index': u'nuage_dpi_probestats_2017_09_14'}}])

Something that already existed? Can you run on your side?

Set them to None?

Reply:
Yes, I was getting this error in some situations too. I will run a debug and detail you on this soon.